### PR TITLE
[Feature] Add page cache metrics

### PR DESCRIPTION
**Pull request by zaorangyang**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/storage/page_cache.cpp [hunk 1]
```diff
@@ -45,8 +51,31 @@ void StoragePageCache::release_global_cache() {
...
+    });
+
+    StarRocksMetrics::instance()->metrics()->register_metric("page_cache_hit_count", &page_cache_hit_count);
+    StarRocksMetrics::instance()->metrics()->register_hook("page_cache_hit_count", []() {
+        page_cache_hit_count.set_value(StoragePageCache::instance()->get_hit_count());
+    });
+
+    StarRocksMetrics::instance()->metrics()->register_metric("page_cache_usage", &page_cache_usage);
+    StarRocksMetrics::instance()->metrics()->register_hook(
+            "page_cache_usage", []() { page_cache_usage.set_value(StoragePageCache::instance()->memory_usage()); });
```
Comment (trueeyu) [inline be/src/storage/page_cache.cpp:67]: Is this computationally expensive? Can I get the value directly from MemTracker?

**Review by trueeyu — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by zaorangyang** [comment]
run starrocks_admit_test

**Comment by zaorangyang** [comment]
@Mergifyio branch-2.3

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-2.4

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-2.4

**Comment by zaorangyang** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**